### ISSUE TITLE
Missing includes in coreclr/src/debug/createdump/

### ISSUE THREAD

**Issue by eroen** [dotnet/coreclr#19722]
Attempting to bootstrap (with buildbootstrapcli.sh) fails with the following:

```
Scanning dependencies of target createdump_lib
[  0%] Building CXX object src/debug/createdump/CMakeFiles/createdump_lib.dir/createdump.cpp.o
[  0%] Building CXX object src/palrt/CMakeFiles/palrt.dir/coguid.cpp.o
[  0%] Building CXX object src/palrt/CMakeFiles/palrt.dir/comem.cpp.o
In file included from /var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/createdump.cpp:5:
In file included from /var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/createdump.h:60:
/var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/crashinfo.h:36:17: error: implicit instantiation of undefined template 'std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >'
    std::string m_coreclrPath;                      // the path of the coreclr module or empty if none
                ^
/usr/include/c++/v1/iosfwd:193:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_string;
                               ^
In file included from /var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/createdump.cpp:5:
In file included from /var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/createdump.h:60:
/var/tmp/dotnet/gentoo-x64/coreclr/src/debug/createdump/crashinfo.h:38:39: error: implicit instantiation of undefined template 'std::__1::array<unsigned long, 34>'
    std::array<elf_aux_val_t, AT_MAX> m_auxvValues; // auxv values
                                      ^
/usr/include/c++/v1/__tuple:223:64: note[  0%] Building CXX object src/palrt/CMakeFiles/palrt.dir/decarith.cpp.o
: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
2 errors generated.
make[2]: *** [src/debug/createdump/CMakeFiles/createdump_lib.dir/build.make:63: src/debug/createdump/CMakeFiles/createdump_lib.dir/createdump.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:61814: src/debug/createdump/CMakeFiles/createdump_lib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Including `<string>` and `<array>` in `coreclr/src/debug/createdump/crashinfo.h` fixes this, but I'm guessing your preferred style puts that elsewhere, seeing as this file doesn't include any of the other headers it obviously requires.

**Comment by mikem8361** [comment]
"createdump.h" is where to add them.  I wonder way building createdump fails with the boot strap script and not the normal build, but it makes sense that these two files be included.

Do you want me to do this?

**Comment by mikem8361** [comment]
I not sure where buildbootstrapcli.sh is to test this. Couldn't find it in the coreclr repo.

**Comment by mikem8361** [comment]
Fix in PR #23075
